<?xml version="1.0"?>
<launch>
    <group ns="arg3">
        <!-- ROS serial control node -->
        <node name="arduino_control" pkg="rosserial_python" type="serial_node.py" required="true">
            <param name="port" value="/dev/ttyACM0"/>
            <param name="baud" value="57600"/>
        </node>
        
        <!-- Collision detection node -->
        <node name="collision_detection_node" pkg="demo3" type="collision_detection_node" required="true" />
        
        <!-- FSM node -->
        <node name="fsm_node" pkg="demo3" type="fsm_node.py" required="true" output="screen">
            <rosparam command="load" file="$(find demo3)/config/demo3.yaml" />
        </node> 
    </group>
</launch>
